lib/siwapp/invoices.ex

```diff
@@ -15,6 +15,10 @@ defmodule Siwapp.Invoices do
     Repo.all(Invoice)
   end
 
+  def list(:preload) do
```

Comment:
I think is better to use the `Ecto.Query.Preload` https://hexdocs.pm/ecto/Ecto.Query.html#preload/3 than the `Repo.preload`, because it will make only 1 query to database.
Besides, instead of `:preload` as parameter I would put `:preload_customer`, because probably at some point we'll need to preload the items or other things.
Summarizing: is better to work with queries, and probably the `Core` (our schema files) is more suitable for this than the context. Read the LiveView book about this and you'll get an idea what I mean.

Reply:
Sorry, not our schema files, but new ones. Look in the book where it explains the Pento.Catalog.Product.Query
Besides, I think we have to make a reorganization of the contexts and folders.